app/scripts/modules/core/search/search.service.ts

```diff
@@ -0,0 +1,52 @@
+import {module} from 'angular';
+import {API_SERVICE, Api} from 'core/api/api.service';
+
+export interface ISearchParams {
+  pageSize?: number;
+  q?: string;
+  type?: string | string[];
+  platform?: string;
+  pageNumber?: number;
+  filter?: {[key: string]: string};
+}
+
+export interface ISearchResults<T> {
+  results: T[];
+  pageNumber?: number;
+  pageSize?: number;
+  platform?: string;
+  query?: string;
+  totalMatches?: number;
+}
+
+export class SearchService {
+
+  static get defaultPageSize() { return 500; };
+
+  static get $inject() { return ['$log', 'API']; }
+
+  constructor(private $log: ng.ILogService, private API: Api) { }
```

Comment:
We've started moving to module imports for `ng.*` types, which, I realize now was never officially communicated.  Sorry about that.  Can you please move them all up to the top?  Thanks.